.gitlab-ci.yml

```diff
@@ -183,9 +181,7 @@ deploy-to-dev:
     - az containerapp revision copy --name portal --resource-group rg-ecc-portal-uks-dev --yaml revision.yml --subscription "Essex County Council (Portal)"
   rules:
     - if: '$CI_COMMIT_BRANCH == "develop"'
-      when: manual
-    - if: '$CI_COMMIT_BRANCH =~ /^release\//'
-      when: manual
+    - when: manual
```

Comment:
Should this be `- when: on_success`?

Reply:
I'm not sure but the intention is:
Run automatically for the develop branch if build-and-push-images job was successful (`on_success` is the default)
Run manually for any other branch (I don't think you can deploy if build-and-push-images failed)

Reply:
Here is the pipeline for this job - deploy to Dev and Preprod are waiting to be run manually.
![image](https://github.com/essexcountycouncil/essex-gov-uk-drupal/assets/56226/83625064-98b4-49a0-aa05-f8450829c967)
When merged, I think the pipeline for _develop_ will deploy to Dev automatically

Reply:
I think I see two things wrong here if the intention is to run automatically on the `develop` branch. The first is that the `- when: manual` part is a list item under `rules:` rather than property of the `- if:` condition. Secondly as you mention that by default the `when:` property is `on_success` the line isn't needed at all anyway.

I was looking at [this bit of gitlab docs originally](https://docs.gitlab.com/ee/ci/yaml/#when), and your SO link was also helpful.

Reply:
With you now... as discussed on the call....